### DISCUSSION TITLE
Probe sensor not working

### DISCUSSION THREAD

**Discussion by obones** [mechano/Kingroon_KP3S_Pro_Marlin#11] [Q&A]
Hello,

As I have updated to a BLTouch enabled sensor, I'm trying to have it work but whenever I ask to home on the Z axis, it stops doing nothing.
In octoprint, I have the following log:

```
Changing monitoring state from "Offline" to "Opening serial connection"
Connecting to port /dev/ttyUSB0, baudrate 115200
Changing monitoring state from "Opening serial connection" to "Connecting"
Connected to: Serial<id=0xaf9ea670, open=True>(port='/dev/ttyUSB0', baudrate=115200, bytesize=8, parity='N', stopbits=1, timeout=10.0, xonxoff=False, rtscts=False, dsrdtr=False), starting monitor
Send: N0 M110 N0*125
Recv: ok
Send: N0 M110 N0*125
Changing monitoring state from "Connecting" to "Operational"
Recv: ok
Send: N0 M110 N0*125
Recv: ok
Send: N1 M115*39
Recv: FIRMWARE_NAME:Marlin bugfix-2.1.x (Dec 29 2023 21:59:08) SOURCE_CODE_URL:github.com/MarlinFirmware/Marlin PROTOCOL_VERSION:1.0 MACHINE_TYPE:KP3S Pro EXTRUDER_COUNT:1 UUID:cede2a2f-41a2-4748-9b12-c55c62f367ff

---------------------------------------------------------------------------
Your printer's firmware is a development build of Marlin (build date
20231229). It might be more unstable than a release version and should be
kept up-to-date.

Learn more at https://faq.octoprint.org/warning-firmware-development
---------------------------------------------------------------------------


Recv: Cap:SERIAL_XON_XOFF:0
Recv: Cap:BINARY_FILE_TRANSFER:0
Recv: Cap:EEPROM:1
Recv: Cap:VOLUMETRIC:1
Recv: Cap:AUTOREPORT_POS:0
Recv: Cap:AUTOREPORT_TEMP:1
Recv: Cap:PROGRESS:0
Recv: Cap:PRINT_JOB:1
Recv: Cap:AUTOLEVEL:1
Recv: Cap:RUNOUT:1
Recv: Cap:Z_PROBE:1
Recv: Cap:LEVELING_DATA:1
Recv: Cap:BUILD_PERCENT:0
Recv: Cap:SOFTWARE_POWER:0
Recv: Cap:TOGGLE_LIGHTS:0
Recv: Cap:CASE_LIGHT_BRIGHTNESS:0
Recv: Cap:EMERGENCY_PARSER:0
Recv: Cap:HOST_ACTION_COMMANDS:0
Recv: Cap:PROMPT_SUPPORT:0
Recv: Cap:SDCARD:1
Recv: Cap:MULTI_VOLUME:0
Recv: Cap:REPEAT:0
Recv: Cap:SD_WRITE:1
Recv: Cap:AUTOREPORT_SD_STATUS:0
Recv: Cap:LONG_FILENAME:0
Recv: Cap:LFN_WRITE:0
Recv: Cap:CUSTOM_FIRMWARE_UPLOAD:0
Recv: Cap:EXTENDED_M20:0
Recv: Cap:THERMAL_PROTECTION:1
Recv: Cap:MOTION_MODES:0
Recv: Cap:ARCS:1
Recv: Cap:BABYSTEPPING:0
Recv: Cap:EP_BABYSTEP:0
Recv: Cap:CHAMBER_TEMPERATURE:0
Recv: Cap:COOLER_TEMPERATURE:0
Recv: Cap:MEATPACK:0
Recv: Cap:CONFIG_EXPORT:0
Recv: ok
Send: M21
Recv: echo:SD card ok
Recv: ok
Send: M155 S2
Recv: ok
Send: M20
Recv: Begin file list
Recv: KK_BAT~1.GCO 6200855
Recv: KK_CUB~1.GCO 695944
Recv: KK_POR~1.GCO 1749779
Recv: KK_TOU~1.GCO 14900796
Recv: KK_8_P~1.GCO 1381673
Recv: T5-4.7/KK_TOP~1.GCO 522592
Recv: T5-4.7/KK_BOT~1.GCO 192320
Recv: T5-4.7/KK_BRA~1.GCO 378517
Recv: T5-4.7/KK_LOW~1.GCO 2084412
Recv: KK_LIT~1.GCO 4110245
Recv: KK_XMA~1.GCO 29429080
Recv: KK_FAN~1.GCO 4985415
Recv: 4015_V~1.GCO 1598844
Recv: SUPPOR~1.GCO 502573
Recv: End file list
Recv: ok
Send: M20
Recv: Begin file list
Recv: KK_BAT~1.GCO 6200855
Recv: KK_CUB~1.GCO 695944
Recv: KK_POR~1.GCO 1749779
Recv: KK_TOU~1.GCO 14900796
Recv: KK_8_P~1.GCO 1381673
Recv: T5-4.7/KK_TOP~1.GCO 522592
Recv: T5-4.7/KK_BOT~1.GCO 192320
Recv: T5-4.7/KK_BRA~1.GCO 378517
Recv: T5-4.7/KK_LOW~1.GCO 2084412
Recv: KK_LIT~1.GCO 4110245
Recv: KK_XMA~1.GCO 29429080
Recv: KK_FAN~1.GCO 4985415
Recv: 4015_V~1.GCO 1598844
Recv: SUPPOR~1.GCO 502573
Recv: End file list
Recv: ok
Recv:  T:19.58 /0.00 B:18.47 /0.00 @:0 B@:0
Recv:  T:19.42 /0.00 B:18.34 /0.00 @:0 B@:0
Recv: echo:busy: processing
Printer seems to support the busy protocol, will adjust timeouts and set busy interval accordingly
Send: M113 S9
Recv:  T:19.49 /0.00 B:18.38 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:19.44 /0.00 B:18.30 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:19.55 /0.00 B:18.43 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:19.59 /0.00 B:18.48 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv:  T:19.37 /0.00 B:18.23 /0.00 @:0 B@:0
Recv: echo:busy: processing
Recv: Error:!! STOP called because of BLTouch error - restart with M999
Recv: Error:Printer stopped due to errors. Fix the error and use M999 to restart. (Temperature is reset. Set it after restarting)
Recv: Error:!! STOP called because of BLTouch error - restart with M999
Recv: X:72.00 Y:95.00 Z:10.80 E:0.00 Count X:11520 Y:15200 Z:8640
Recv: echo:Home Z First
Recv: ok
Recv:  T:19.57 /0.00 B:18.52 /0.00 @:0 B@:0
Recv: ok
Recv:  T:19.56 /0.00 B:18.46 /0.00 @:0 B@:0
Recv:  T:19.66 /0.00 B:18.45 /0.00 @:0 B@:0
Recv: Error:!! STOP called because of BLTouch error - restart with M999
Recv:  T:19.37 /0.00 B:18.38 /0.00 @:0 B@:0
Recv: X:72.00 Y:95.00 Z:10.80 E:0.00 Count X:11520 Y:15200 Z:11840
Recv: ok
Recv: echo:endstops hit:  P:14.80
Recv:  T:19.61 /0.00 B:18.48 /0.00 @:0 B@:0
```

So, apparently there is an error with the probe but I can't seem to figure out what's wrong.
I have bought a sensor from Triangle lab and mounted it like this:

![image](https://github.com/mechano/Kingroon_KP3S_Pro_Marlin_Firmware/assets/1314739/0f48c884-6485-4b5f-9dfc-b12d50f31277)

And the probe comes out and back in a few times when the printer starts, or even if I use the touch panel to do a self test on it.

I have tried reverting the black and white wires to no avail. 
Would you have any suggestion as to what I did wrong?

**Comment by mechano** [comment]
I think the problem was that the function HOST_ACTION_COMMANDS was not active. It's needed by Octoprint. 

Now it's active and I've uploaded new firmware build.

**Reply by obones** [reply]
Sadly, it's not related. I did update to the latest firmware (thanks for that) and no longer have the associated warning but this does not solve this issue.
Looking at Marlin issues/support, it appears that it may be related to my wiring so I'll verify it

**Comment by mechano** [comment]
90% 3D Touch problems are wiring related.

Right behaviour is:
switch on printer, the needle probes twice, the Z moves up 1cm, the probe
is deployed again and the Z moves slowly to the bed, the needle touch the
bed and is retired the Z moves 5mm up,probe is deployed again and touch the
bed again, Z moves up 10mm. Z homing is complete.

Every different behaviour is a wiring problem.

Il giorno mar 9 gen 2024 alle ore 10:55 obones ***@***.***>
ha scritto:

> Sadly, it's not related. I did update to the latest firmware (thanks for
> that) and no longer have the associated warning but this does not solve
> this issue.
> Looking at Marlin issues/support, it appears that it may be related to my
> wiring so I'll verify it
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mechano/Kingroon_KP3S_Pro_Marlin_Firmware/discussions/11#discussioncomment-8062925>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAV5ZAZ7PQRTIDY2DW3OYPDYNUHY3AVCNFSM6AAAAABBRTQXNWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DANRSHEZDK>
> .
> You are receiving this because you commented.Message ID:
> <mechano/Kingroon_KP3S_Pro_Marlin_Firmware/repo-discussions/11/comments/8062925
> @github.com>
>


-- 
Roberto Giuffrè
tel. +39-328-9140378
email: ***@***.***
pec: ***@***.***

**Comment by obones** [comment]
Ok, I finally got it to work after having built my own firmware for testing various options, just to figure out that the sensor was not plugged in the right port.
The picture in #1 says this:

> connect white/black to where zmin was

But that's not correct, what must be done is what the text is saying:

> For this firmware unplug the actual Z Min switch, leave unplugged.
And connect the Z Min of the BL Touch to the Z Max fre green socket.

So to sum up, the correct setup is this:

1. Plug the Orange/Red/Brown connector from the probe to the BLTouch connector of the motherboard, Brown is G.
1. Unplug Z wire from the Z Min connector on the motherboard
1. Plug the White/Black connector from the probe to the Z Max connector of the motherboard, White wire close to edge of the board, Black wire in the middle, third pin is left floating.

Sorry for all the noise, I should have read more carefully.
